### PR TITLE
[GEOS-8336][GEOS-8337] - Monitoring Hibernate obsolete dependencies / Monitoring Hibernate properties causing issues with the latest spring mvc changes

### PR DESCRIPTION
**Pull request by afabiani**

### CODE REVIEW

File: src/extension/monitor/hibernate/pom.xml [hunk 1]
```diff
@@ -11,50 +11,71 @@
...
-   <scope>test</scope>
-  </dependency>
+	</dependencies>
 
- </dependencies>
-
- <properties>
-   <test.exclude.pattern>**/HibernateMonitorDAOTest.*</test.exclude.pattern>
- </properties>
+	<properties>
```
Comment (aaime) [inline src/extension/monitor/hibernate/pom.xml]: This exclusion can be removed, there is a single test class and it has a different name: HibernateMonitorDAO2Test

File: src/extension/monitor/hibernate/pom.xml [hunk 2]
```diff
@@ -11,50 +11,71 @@
...
+	<groupId>org.geoserver.extension</groupId>
+	<artifactId>gs-monitor-hibernate</artifactId>
+	<packaging>jar</packaging>
+	<name>Monitor Hibernate Extension</name>
 
- <groupId>org.geoserver.extension</groupId>
- <artifactId>gs-monitor-hibernate</artifactId>
- <packaging>jar</packaging>
- <name>Monitor Hibernate Extension</name>
+	<dependencies>
```
Comment (aaime) [inline src/extension/monitor/hibernate/pom.xml]: Files should be indented with spaces (as the original version was)

File: src/extension/monitor/hibernate/src/main/resources/org/geoserver/monitor/hib/db.properties [hunk 3]
```diff
@@ -20,3 +20,5 @@ url=jdbc:h2:file:${GEOSERVER_DATA_DIR}/monitoring/monitoring
 #url=jdbc:oracle:thin:@[host]:[port]\:[instance]
 #username=[username]
 #password=[password]
+
+gwc.context.suffix=gwc
```
Comment (aaime) [inline src/extension/monitor/hibernate/src/main/resources/org/geoserver/monitor/hib/db.properties]: For this one I'd rather fix the source issue (more on this later, let me have a look)

File: src/extension/monitor/hibernate/src/main/resources/applicationContext-hib2.xml [hunk 4]
```diff
@@ -1,27 +1,36 @@
...
-        </props>
-      </property>
+   <bean id="monitoring-hibPropertyConfigurer" class="org.geoserver.config.GeoServerPropertyConfigurer">
+
+        <property name="order" value="100"/>
+        
+        <constructor-arg ref="dataDirectory"/>
+
+        <property name="ignoreResourceNotFound" value="true" />
+        <property name="ignoreUnresolvablePlaceholders" value="true" />
```
Comment (aaime) [inline src/extension/monitor/hibernate/src/main/resources/applicationContext-hib2.xml]: I was thinking exactly of these properties to solve the GWC property configurer nuke... did you already try it by any chance?

**Comment by aaime** [comment]
Hi Alessio, I've tested the change and I got it working (full startup) without any other changes, and my property config files do not contain the gwc property. 
I am wondering if I just go lucky with classpath, or does it match your experience?

**Comment by afabiani** [comment]
@aaime deps here are quite old and for sure conflicting with ones from other plugins, like GeoFence and GeoFence Server.

File: src/extension/monitor/hibernate/src/main/resources/org/geoserver/monitor/hib/db.properties (hunk 3 above)
Comment (aaime) [reply src/extension/monitor/hibernate/src/main/resources/org/geoserver/monitor/hib/db.properties]: Can we remove this property? Should still work

**Pull request merged**